<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2018-08-06" />

<title>Install organisms from Ensembl</title>

<script src="install_organisms_from_ensembl_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="install_organisms_from_ensembl_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="install_organisms_from_ensembl_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="install_organisms_from_ensembl_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="install_organisms_from_ensembl_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="install_organisms_from_ensembl_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="install_organisms_from_ensembl_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="install_organisms_from_ensembl_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="install_organisms_from_ensembl_files/navigation-1.1/tabsets.js"></script>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Install organisms from Ensembl</h1>
<h4 class="author"><em>Jacques van Helden</em></h4>
<h4 class="date"><em>2018-08-06</em></h4>

</div>


<div id="installing-genomes-and-variations-from" class="section level1">
<h1><span class="header-section-number">1</span> Installing genomes and variations from</h1>
<p><a href="#sect:install_ensembl_genome"><span class="math display">\[sect:install\_ensembl\_genome\]</span></a></p>
<p>includes a series of programs to download and install genomes from Ensembl.</p>
<ol style="list-style-type: decimal">
<li><p><code>install-ensembl-genome</code> is a wrapper enabling to autmoatize the download (genome sequences, features, variations) and configuration tasks.</p></li>
<li><p>`` downloads the genomics sequences and converts them in the raw format required for .</p></li>
<li><p><code>download-ensembl-features</code> downloads tab-delimited text files describing genomic features (transcripts, CDS, genes, …).</p></li>
<li><p><code>download-ensembl-variations</code> downloads tab-delimited text files describing genomic variations (polymorphism).</p></li>
</ol>
<div id="installing-genomes-from-ensembl" class="section level2">
<h2><span class="header-section-number">1.1</span> Installing genomes from Ensembl</h2>
<p>The program <code>install-ensembl-genome</code> manages all the required steps to download and install a genome (sequence, features, and optionally variations) from Ensembl to .</p>
<p>It performs the following tasks:</p>
<ol style="list-style-type: decimal">
<li><p>The option runs the program to download the complete genomic sequence of a given organism from the Web site, and formats it according to requirements (conversion from the original fasta sequence file to one file per chromosome, in raw format).</p></li>
<li><p>The option runs to download the positions and descriptions of genomic features (genes, CDS, mRNAs, …).</p></li>
<li><p>Optionally, when the option is activated, run to download the description of genomic variations (polymorphism). Note that variations are supported only for a subset of genomes.</p></li>
<li><p>Update configuration () to make the genome available to other programs in the current site.</p></li>
<li><p>Run the additional tasks () required to have a fully functional genome on the local site: compute genomic statisics (intergenic sizes, …) and background models (oligonucleotide and dyad frequencies).</p></li>
<li><p>With the option , the program returns the list species available on the Ensembl server, together with their status of availability for the 3 data types (genome sequence, features, variations). When this option is called, the program does not install any genome.</p></li>
</ol>
<p>The detailed description of the program and the list of options can be obtained with the option .</p>
<pre><code>## Get the description of the program + all options
install-ensembl-genome -help</code></pre>
<div id="getting-the-list-of-available-genomes" class="section level3">
<h3><span class="header-section-number">1.1.1</span> Getting the list of available genomes</h3>
<p>Before installing a genome, it is generally a good idea to know which genomes are available. For this, use the option .</p>
<pre><code>## Retrieve the list of supported species on EnsEMBL
install-ensembl-genome -v 1  -available_species \
  -o available_species_ensembl.tab

## Read the result file
more available_species_ensembl.tab</code></pre>
<p><em>Note:</em> inter-individual variations are available for a subset only of the genomes available in . The option indicates, for each species, the availability (genome, features, variations). Obviously, the programs to analyse regulatory variations (, , ) are working only for the genomes documented with variations.</p>
</div>
<div id="installing-a-genome-from-ensembl" class="section level3">
<h3><span class="header-section-number">1.1.2</span> Installing a genome from Ensembl</h3>
<p>We can now download and install the complete genomic sequence for the species of our choice. For the sake of space and time economy, we will use a small genome for this manual: the budding yeast .</p>
<p><em>Beware</em>: some installation steps take a lot of time. For large genomes (e.g. Vertebrate organisms), the full installation can thus take several hours. This should in principle not be a big issue, since installing a genome is not a daily task, but it is worth knowing that the whole process requires a continuous connection during several hours.</p>
<pre><code>## Install the genome sequences for a selected organism
install-ensembl-genome -v 2 -species Saccharomyces_cerevisiae</code></pre>
<p>This command will automatically run all the installation tasks described above, except the installation of variations (see Section <a href="#sect:download_ensembl_variations">1.3</a>).</p>
</div>
</div>
<div id="installing-genomes-from-ensemblgenomes" class="section level2">
<h2><span class="header-section-number">1.2</span> Installing genomes from EnsemblGenomes</h2>
<p>The historical project was focused on vertebrate genomes + a few model organisms (, , …).</p>
<p>A more recent project called extends the project to a wider taxonomic range (in Oct 2014, there are &gt;15,000 genomes available at EnsemblGenomes, where as Ensembl only provides 69 genomes).</p>
<p>The program supports the installation of genomes from as well as . By default, it opens a connection to the historical database, but the option enables to install genomes from the new project .</p>
<pre><code>## Get the list of available species from the extended project
## EnsemblGenomes
install-ensembl-genome -v 2 -available_species -db ensemblgenomes \
   -o available_species_at_EnsemblGenome.txt</code></pre>
<p>You can then identify your genome of interest in the file , and start the installation (don’t forget the option .</p>
<pre><code>## Install Escherichia coli (strain K12 MG1665) from EnsemblGenomes
install-ensembl-genome -v 2 -db ensemblgenomes \
   -species Escherichia_coli_str_k_12_substr_mg1655</code></pre>
</div>
<div id="downloading-variations" class="section level2">
<h2><span class="header-section-number">1.3</span> Downloading variations</h2>
<p>The program downloads variations from the Web site, and installs it on the local site.</p>
<p>This program relies on , which must be installed beforehand on your computer.</p>
<pre><code>## Retrieve the list of supported species in the EnsEMBL variation database
download-ensembl-variations -v 1  -available_species</code></pre>
<p>We can now download all the variations available for the yeast.</p>
<pre><code>## Download all variations for a selected organism on your server
download-ensembl-variations -v 1 -species Saccharomyces_cerevisiae</code></pre>
<p>Variation files are stored in a specific subfolder for the specified organism.</p>
<pre><code>## Check the content of the variation directory for the yeast
make -f makefiles/variation-scan_demo.mk \
   SPECIES=Saccharomyces_cerevisiae ASSEMBLY=R64-1-1 \
   variation_stats</code></pre>
<p>This command will indicate the location of the variation directory on your server, and count the number of lines for each variation file (there is one separate file per chromosome or contig).</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
