<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2019-02-08" />

<title>Install organisms from NCBI</title>

<script src="install_organisms_from_ncbi_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="install_organisms_from_ncbi_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="install_organisms_from_ncbi_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="install_organisms_from_ncbi_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="install_organisms_from_ncbi_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="install_organisms_from_ncbi_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="install_organisms_from_ncbi_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="install_organisms_from_ncbi_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="install_organisms_from_ncbi_files/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #ffffff; color: #1f1c1b; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; background-color: #ffffff; color: #a0a0a0; border-right: 1px solid #a0a0a0; }
td.sourceCode { padding-left: 5px; }
pre, code { color: #1f1c1b; background-color: #ffffff; }
code > span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code > span.dt { color: #0057ae; } /* DataType */
code > span.dv { color: #b08000; } /* DecVal */
code > span.bn { color: #b08000; } /* BaseN */
code > span.fl { color: #b08000; } /* Float */
code > span.cn { color: #aa5500; } /* Constant */
code > span.ch { color: #924c9d; } /* Char */
code > span.sc { color: #3daee9; } /* SpecialChar */
code > span.st { color: #bf0303; } /* String */
code > span.vs { color: #bf0303; } /* VerbatimString */
code > span.ss { color: #ff5500; } /* SpecialString */
code > span.im { color: #ff5500; } /* Import */
code > span.co { color: #898887; } /* Comment */
code > span.do { color: #607880; } /* Documentation */
code > span.an { color: #ca60ca; } /* Annotation */
code > span.cv { color: #0095ff; } /* CommentVar */
code > span.ot { color: #006e28; } /* Other */
code > span.fu { color: #644a9b; } /* Function */
code > span.va { color: #0057ae; } /* Variable */
code > span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code > span.op { color: #1f1c1b; } /* Operator */
code > span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code > span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code > span.pp { color: #006e28; } /* Preprocessor */
code > span.at { color: #0057ae; } /* Attribute */
code > span.re { color: #0057ae; } /* RegionMarker */
code > span.in { color: #b08000; } /* Information */
code > span.wa { color: #bf0303; } /* Warning */
code > span.al { color: #bf0303; font-weight: bold; } /* Alert */
code > span.er { color: #bf0303; text-decoration: underline; } /* Error */
code > span. { color: #1f1c1b; } /* Normal */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Install organisms from NCBI</h1>
<h4 class="author"><em>Jacques van Helden</em></h4>
<h4 class="date"><em>2019-02-08</em></h4>

</div>


<div id="introduction" class="section level1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>The goal of this tutorial is to explain how to install genomes from NCBI in your RSAT server.</p>
</div>
<div id="prerequisite" class="section level1">
<h1><span class="header-section-number">2</span> Prerequisite</h1>
<p>This protocol assumes that you have a working instance of the RSAT server, and that you are familar with the Unix terminal.</p>
</div>
<div id="protocol" class="section level1">
<h1><span class="header-section-number">3</span> Protocol</h1>
<p>We illustrate the installation procedure with a bacterial genome, since NCBI is the main source of prokaryote genomes for RSAT (<a href="http://prokaryotes.rsat.eu/" class="uri">http://prokaryotes.rsat.eu/</a>). The same procedure can however apply to other taxa, as far as they are available on the NCBI FTP server (<a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/" class="uri">ftp://ftp.ncbi.nlm.nih.gov/genomes/</a>).</p>
<div id="data-source" class="section level2">
<h2><span class="header-section-number">3.1</span> Data source</h2>
<p>Genomes are downloaded from the refseq repository: <a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq" class="uri">ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq</a></p>
</div>
<div id="getting-the-list-of-available-species" class="section level2">
<h2><span class="header-section-number">3.2</span> Getting the list of available species</h2>
<p>The first step to perform with the command <code>ìnstall-organisms</code> is to get the list of organisms available for a given taxonomic group.</p>
<p>The NCBI FTP genome site is organised by “main” <strong>taxonomic groups</strong>:</p>
<ul>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/archaea/">archaea</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/">bacteria</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/fungi/">fungi</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/invertebrate/">invertebrate</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/plant/">plant</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/mitochondrion/">mitochondrion</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/plasmid">plasmid</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/plastid">plastid</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/protozoa/">protozoa</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/">vertebrate_mammalian</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_other/">vertebrate_other</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/viral/">viral</a></li>
</ul>
<p>The option <code>-task available</code> returns the complete list of available species for the specified group.</p>
<p>The command below gets the list of all available bacteria and stores them in a text file. It can easily be adapted to another group by changing the environment variable <code>GROUP</code>.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Get curret date to use as suffix for output files</span>
<span class="bu">export</span> <span class="va">TODAY=</span><span class="kw">`</span><span class="fu">date</span> +%Y-%m-%d<span class="kw">`</span>

<span class="co">## Choose a taxonomic group of interest</span>
<span class="bu">export</span> <span class="va">GROUP=</span>bacteria

<span class="co">## Get the list of bacterial species available on the NCBI FTP site</span>
<span class="ex">install-organism</span> -v 1 -group <span class="va">${GROUP}</span> -task available \
  -o available_<span class="va">${GROUP}</span>_<span class="va">${TODAY}</span>.txt
  
<span class="co">## Count the number of species.</span>
<span class="co">## Note: grep is used to filter out the comment lines.</span>
<span class="fu">grep</span> -v <span class="st">&#39;^;&#39;</span> available_<span class="va">${GROUP}</span>_<span class="va">${TODAY}</span>.txt <span class="kw">|</span> <span class="fu">wc</span> -l</code></pre></div>
<p>We can of course do the same for other taxa, for example Archaea, Fungi or Mammalian.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="bu">export</span> <span class="va">NCBI_GROUPS=</span><span class="st">&#39;archaea bacteria fungi invertebrate plant protozoa vertebrate_mammalian vertebrate_other viral&#39;</span>

<span class="bu">echo</span> <span class="st">&quot;NCBI_GROUPS </span><span class="va">${NCBI_GROUPS}</span><span class="st">&quot;</span>
<span class="kw">for</span> <span class="ex">ncbi_group</span> in <span class="va">$NCBI_GROUPS</span><span class="kw">;</span> <span class="kw">do</span> <span class="kw">\</span>
  <span class="bu">echo</span> <span class="st">&quot;Getting list of NCBI species for </span><span class="va">${ncbi_group}</span><span class="st">&quot;</span> <span class="kw">;</span> <span class="kw">\</span>
  <span class="ex">install-organism</span> -v 1 -group <span class="va">${ncbi_group}</span> -task available \
    -o available_<span class="va">${ncbi_group}</span>_<span class="va">${TODAY}</span>.txt <span class="kw">;</span> <span class="kw">\</span>
<span class="kw">done</span>


<span class="co">## Approx count of species per taxa</span>
<span class="co">## (each file starts with 15 comment lines, which are be substracted from each count)</span>
<span class="fu">wc</span> -l available_*_<span class="va">${TODAY}</span>.txt <span class="kw">|</span>  <span class="fu">sort</span> -nr  <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{print &quot;|\t&quot;$1-15&quot;\t| &quot;$2&quot;\t|&quot;}&#39;</span> </code></pre></div>
<div id="number-of-species-per-group" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Number of species per group</h3>
<p>Here is the result on Feb 1, 2018</p>
<table>
<thead>
<tr class="header">
<th>Number of species</th>
<th>Taxonomic group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>32109</td>
<td>total</td>
</tr>
<tr class="even">
<td>22481</td>
<td>available_bacteria_2019-02-08.txt</td>
</tr>
<tr class="odd">
<td>7956</td>
<td>available_viral_2019-02-08.txt</td>
</tr>
<tr class="even">
<td>621</td>
<td>available_archaea_2019-02-08.txt</td>
</tr>
<tr class="odd">
<td>289</td>
<td>available_fungi_2019-02-08.txt</td>
</tr>
<tr class="even">
<td>179</td>
<td>available_invertebrate_2019-02-08.txt</td>
</tr>
<tr class="odd">
<td>149</td>
<td>available_vertebrate_other_2019-02-08.txt</td>
</tr>
<tr class="even">
<td>123</td>
<td>available_vertebrate_mammalian_2019-02-08.txt</td>
</tr>
<tr class="odd">
<td>111</td>
<td>available_plant_2019-02-08.txt</td>
</tr>
<tr class="even">
<td>80</td>
<td>available_protozoa_2019-02-08.txt</td>
</tr>
</tbody>
</table>
<p><strong>Notes</strong></p>
<ul>
<li><p>The <strong>number of available genomes</strong> exceeds by far the number of species, because some species are represented by different strains, in particular for bacteria (for some bacterial species of interest for health, several thousands of strains have been sequenced).</p></li>
<li><p>Some of these genomes are only partly sequenced, or poorly assembled, and the level of annotation is quite variable. NCBI subdivided these available genomes in 4 categories depending on the level of completion of the sequencing project.</p></li>
</ul>
</div>
</div>
<div id="installing-the-genome-for-a-given-species" class="section level2">
<h2><span class="header-section-number">3.3</span> Installing the genome for a given species</h2>
<div id="listing-the-available-strains-for-the-species-of-interest" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Listing the available strains for the species of interest</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## List the available strains for a given species (e.g. Rhodospirillum centenum).</span>
<span class="ex">install-organism</span> -v 1 -group bacteria \
  -species <span class="va">${SPECIES}</span> \
  -task list</code></pre></div>
<p>This command actually lists the folders.</p>
</div>
<div id="downloading-the-genomes" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Downloading the genome(s)</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Select a species and straint of interest </span>
<span class="co">## (can be adapted to your needs)</span>
<span class="bu">export</span> <span class="va">SPECIES=</span>Rhodospirillum_centenum
<span class="bu">export</span> <span class="va">STRAIN=</span>GCF_000016185.1_ASM1618v1
<span class="bu">export</span> <span class="va">TAXON=</span>Rhodospirillum

<span class="co">## An example of alternative set of parameters</span>
<span class="co"># export SPECIES=Rhizobium_etli</span>
<span class="co"># export STRAIN=GCF_000092045.1_ASM9204v1</span>
<span class="co"># export TAXON=Rhizobiales</span>

<span class="co">## Yet another example, just for Mere</span>
<span class="co"># export SPECIES=Sinorhizobium_meliloti</span>
<span class="co"># export STRAIN=GCF_000006965.1_ASM696v1</span>
<span class="co"># export TAXON=Rhizobiales</span>

<span class="co">## Download the genome(s) for a given species </span>
<span class="co">## (e.g. Rhodospirillum_centenum)  </span>
<span class="ex">install-organism</span> -v 1 -group bacteria \
  -species <span class="va">${SPECIES}</span> \
  -task download,list</code></pre></div>
<p><strong>Notes</strong></p>
<ol style="list-style-type: decimal">
<li><p>The option <code>-species</code> requires to specify the Genus (with a leading uppercase) and species (in lowercases) separated by an undescore character (<code>_</code>), with exactly the same spelling and cases as on the NCBI FTP refseq site.</p></li>
<li><p>Downloaded genomes will be installed in a folder <code>$RSAT/downloads/refseq</code> on your computer. Each strain is stored in a separate folder, with the same organistion as on the NCBI FTP site.</p></li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ls <span class="op">-</span><span class="dv">1</span> <span class="op">$</span>RSAT<span class="op">/</span>downloads<span class="op">/</span>refseq<span class="op">/</span>bacteria<span class="op">/</span><span class="er">$</span>{SPECIES}<span class="op">/</span>latest_assembly_versions</code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li><strong>Beware:</strong> for some bacterial species of particular interest (e.g. important human pathogens), severak thousands of strains have been sequenced. It might cost a lot of time and disk space to download all of them. TO avoid that, the option <code>-strain</code> enables to specify the full strain name (more precisely the subfolder corresponding to the strain of interest in the species folder on the NCBI FTP site).</li>
</ol>
</div>
<div id="parsing-a-selected-genome-from-ncbi" class="section level3">
<h3><span class="header-section-number">3.3.3</span> Parsing a selected genome from NCBI</h3>
<p>Parsing consists in extracting the information from a text-formatted file, in order to organise it in a computable structure.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Parse a given strain of a given genome </span>
<span class="co">## (e.g. strain GCF_000016185.1_ASM1618v1 of Rhodospirillum centenum) </span>
<span class="ex">install-organism</span> -v 2 -group bacteria \
  -species <span class="va">${SPECIES}</span> \
  -strain <span class="va">${STRAIN}</span> \
  -task parse</code></pre></div>
<p><strong>Note:</strong> if the option <code>-strain</code> is omitted, <code>install-organisms</code> parses all the strains found in the download folder of the selected organism.</p>
<p>For each parsed genome, the program creates a folder whose name is the concatenation of the genus, species and strain names: <code>$RSAT/public_html/data/[Genus_species_strain]</code></p>
<p>From now on we will directly refer to this concatenated name as the “Organism name” (generally denoted by the option <code>-org</code> in RSAT commands.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Check the parsing result</span>
<span class="fu">ls</span> -l <span class="va">${RSAT}</span>/public_html/data/genomes/<span class="va">${SPECIES}</span>_<span class="va">${STRAIN}</span>/genome/

<span class="co">## Measure the disk space occupied by this genome (note: this will be increased in the subsequent steps)</span>
<span class="fu">du</span> -sm <span class="va">${RSAT}</span>/public_html/data/genomes/<span class="va">${SPECIES}</span>_<span class="va">${STRAIN}</span>/genome/</code></pre></div>
<p>This gives the number of megabases occupied by the genome folder.</p>
</div>
<div id="adding-the-parsed-genome-to-the-list-of-supported-genomes" class="section level3">
<h3><span class="header-section-number">3.3.4</span> Adding the parsed genome to the list of supported genomes</h3>
<p>In order to make the parsed genome available for RSAT, you need to run the configuration task.</p>
<p>Note that for the subsequent tasks, the options <code>-species</code> and <code>-strains</code> are not used anymore, we will instead specify “organisms” installed on RSAT, identified by the concatenation of Genus, species and strain.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Configure the organism for RSAT</span>
<span class="ex">install-organism</span> -v 1 -org <span class="va">${SPECIES}</span>_<span class="va">${STRAIN}</span> -task config


<span class="co">## Check that the organism has weel been declared</span>
<span class="ex">supported-organisms</span> -taxon <span class="va">${TAXON}</span></code></pre></div>
</div>
<div id="finalizing-the-installation" class="section level3">
<h3><span class="header-section-number">3.3.5</span> Finalizing the installation</h3>
<p>Several more tasks need to be performed before we can consider this organism as “installed”.</p>
<p>The simplest way to run it is to select the “default” tasks.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Run the installation task (retrieve upstream sequences compute background oligo and dyad frequencies, index fasta genome for bedtols, ...)</span>

<span class="ex">install-organism</span> -v 1 -org <span class="va">${SPECIES}</span>_<span class="va">${STRAIN}</span> -task default</code></pre></div>
<div id="detailed-list-of-supported-installation-tasks-default-and-others" class="section level4">
<h4><span class="header-section-number">3.3.5.1</span> Detailed list of supported installation tasks (default and others)</h4>
<table style="width:89%;">
<colgroup>
<col width="19%" />
<col width="69%" />
</colgroup>
<thead>
<tr class="header">
<th>Task</th>
<th>Meaning</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>default</td>
<td>run all the default tasks useful for the installation of a genome (this is a subset of the tasks specified below)</td>
</tr>
<tr class="even">
<td>start_stop</td>
<td>compute trinucleotide frequencies in all the start and stop codons (validation of the correspondance between genome sequences and gene coordinates)</td>
</tr>
<tr class="odd">
<td>allup</td>
<td>Retrieve all upstream sequences in order to compute the background models (oligonucleotide and dyad frequencies)</td>
</tr>
<tr class="even">
<td>seq_len_distrib</td>
<td>Compute the distribution of upstream sequences</td>
</tr>
<tr class="odd">
<td>genome_segments</td>
<td>SPlit the genome into segments corresponding to different genomic region types: genic, intergenic, …</td>
</tr>
<tr class="even">
<td>upstream_freq</td>
<td>Compute oligonucleotide or/and dyad frequencies in upstream sequences (must be combined with tasks oligos and/or dyads)</td>
</tr>
<tr class="odd">
<td>genome_freq</td>
<td>Compute oligonucleotide or/and dyad frequencies in the whole genome (must be combined with tasks oligos and/or dyads)</td>
</tr>
<tr class="even">
<td>protein_freq</td>
<td>Compute amino acid, di- and tri-peptide frequencies in all protein sequences</td>
</tr>
<tr class="odd">
<td>protein_len</td>
<td>Compute distribution of protein lengths</td>
</tr>
<tr class="even">
<td>oligos</td>
<td>Compute oligonucleotide frequencies (size 1 to 8) in a given type of sequences. Must be combined with stasks upstream or genomic</td>
</tr>
<tr class="odd">
<td>dyads</td>
<td>Idem for dyads, i.e. spaced pairs of trinucleotides</td>
</tr>
<tr class="even">
<td>fasta_genome</td>
<td>convert genome sequence in fasta in order to make it usable by bedtools</td>
</tr>
<tr class="odd">
<td>chrom_sizes</td>
<td>compute chromosome sizes from genomic sequences</td>
</tr>
<tr class="even">
<td>index_bedtools</td>
<td>index genome to enable its ultra-fast processing by bedtools</td>
</tr>
<tr class="odd">
<td>uninstall</td>
<td>uninstall a genome, i.e. stop displaying it as available. Beware, this does not erase the genome folder, for this you can use the option <code>-task erase</code></td>
</tr>
<tr class="even">
<td>erase</td>
<td>Erase the whole folder of your organism in <code>$RSAT/public_html/data</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="checking-that-the-organism-is-well-supported" class="section level3">
<h3><span class="header-section-number">3.3.6</span> Checking that the organism is well supported</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Get all the organisms of the Genus Rhodospirillum</span>
<span class="ex">supported-organisms</span> -taxon <span class="va">${TAXON}</span></code></pre></div>
</div>
<div id="checking-the-correctness-of-gene-annotations-by-computing-startstop-codon-frequencies" class="section level3">
<h3><span class="header-section-number">3.3.7</span> Checking the correctness of gene annotations by computing start/stop codon frequencies</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Retrieve all the start codon sequences</span>
<span class="ex">retrieve-seq</span> -all -org <span class="va">${SPECIES}</span>_<span class="va">${STRAIN}</span> -from 0 -to 2 -o start_seq.fasta

<span class="fu">more</span> start_seq.fasta

<span class="co">## Check that most genes have ATG as start codon</span>
<span class="ex">oligo-analysis</span> -v 1 -i start_seq.fasta  -l 3 -1str -return occ,freq -sort <span class="kw">|</span> <span class="fu">more</span></code></pre></div>
<p>You can do the similar test with the stop codon frequencies. Note that nothing forces you to store the stop codong sequences in a local file, you can use the pipe symbol <code>|</code> to concatenate two commands.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Compute all trinucleotide frequencies in the stop codons</span>
<span class="ex">retrieve-seq</span> -all -org <span class="va">${SPECIES}</span>_<span class="va">${STRAIN}</span> \
   -type downstream -from -1 -to -3 -o stop_seq.fasta

<span class="fu">head</span> -20 stop_seq.fasta

<span class="co">## Check that most genes have ATG as start codon</span>
<span class="ex">oligo-analysis</span> -v 1 -i stop_seq.fasta  -l 3 -1str \
   -return occ,freq -sort <span class="kw">|</span> <span class="fu">more</span></code></pre></div>
<hr />
</div>
</div>
<div id="selecting-organisms-with-complete-genomes-or-chromosomes" class="section level2">
<h2><span class="header-section-number">3.4</span> Selecting organisms with Complete genomes or Chromosomes</h2>
<p>For each taxonomic group, NCBI provides a summary file with indications of the sequencing status for each genome project. Here is the link for Bacteria (you can easily adapt it for other groups).</p>
<ul>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt" class="uri">ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt</a></li>
</ul>
<p>This table contains many columns with valuable information for genome management. The column contain is indicated in a separate file.</p>
<ul>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt" class="uri">ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt</a></li>
</ul>
<p>Genome assemblies are classigied in four levels (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4702866/">PMCID: PMC4702866</a>).</p>
<ol style="list-style-type: decimal">
<li><strong>Contig:</strong> Assemblies that include only contigs.</li>
<li><strong>Scaffold:</strong> Includes both scaffolds and contigs.</li>
<li><strong>Chromosome:</strong> Includes chromosome or linkage groups, plus scaffolds and contigs.</li>
<li><strong>Complete genome:</strong> Assemblies for which all molecules are fully sequenced.</li>
</ol>
<p>We can use this assembly summary to select only those genomes with the a completion level that we consider appropriate.</p>
<p>For example, for the RSAT prokaryote server (<a href="http://prokaryote.rsat.eu" class="uri">http://prokaryote.rsat.eu</a>), we restrict the installation to genomes with at least one fully sequenced chromosome (categories “Chromosome” and “Complete Genome”).</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## Get current date to use as suffix for output files</span>
<span class="bu">export</span> <span class="va">TODAY=</span><span class="kw">`</span><span class="fu">date</span> +%Y-%m-%d<span class="kw">`</span>

<span class="co">## Choose a taxonomic group of interest</span>
<span class="bu">export</span> <span class="va">GROUP=</span>bacteria

<span class="co">## Set a file name for the assembly file</span>
<span class="va">ASSEMBLY_URL=</span>ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/<span class="va">${GROUP}</span>/assembly_summary.txt
<span class="va">ASSEMBLY=</span>ncbi_organisms_assembly_summary_<span class="va">${GROUP}</span>_refseq_<span class="va">${TODAY}</span>.tsv
<span class="va">COMPLETE=</span>ncbi_organisms_assembly_summary_<span class="va">${GROUP}</span>_complete_<span class="va">${TODAY}</span>.tsv
<span class="va">ONECHROM=</span>ncbi_organisms_assembly_summary_<span class="va">${GROUP}</span>_chromosome_<span class="va">${TODAY}</span>.tsv
<span class="va">ONECHROM_OR_COMPLETE=</span>ncbi_organisms_assembly_summary_<span class="va">${GROUP}</span>_chromosome_or_complete_<span class="va">${TODAY}</span>.tsv

<span class="co">## Dowload the assembly summary from NCBI FTP site</span>
<span class="fu">wget</span> -nc <span class="va">${ASSEMBLY_URL}</span> --output-document <span class="va">${ASSEMBLY}</span> 

<span class="co">## Count the number of organisms (remove comments first) </span>
<span class="fu">grep</span> -v <span class="st">&#39;^#&#39;</span> <span class="va">${ASSEMBLY}</span> <span class="kw">|</span> <span class="fu">wc</span> -l

<span class="co">## Select only the genomes with a Complete or Chromosome annotation in the field assembly_level</span>
<span class="fu">awk</span> -F <span class="st">&quot;\t&quot;</span> <span class="st">&#39;$12==&quot;Chromosome&quot; &amp;&amp; $11==&quot;latest&quot;{print $0}&#39;</span> \
      <span class="va">${ASSEMBLY}</span> \
      <span class="op">&gt;</span> <span class="va">${ONECHROM}</span>
      
<span class="fu">awk</span> -F <span class="st">&quot;\t&quot;</span> <span class="st">&#39;$12==&quot;Complete Genome&quot; &amp;&amp; $11==&quot;latest&quot;{print $0}&#39;</span> \
      <span class="va">${ASSEMBLY}</span> \
      <span class="op">&gt;</span> <span class="va">${COMPLETE}</span>


<span class="co">## Merge summary tables of species with at least one sequenced chromosome</span>
<span class="fu">cat</span> <span class="va">${COMPLETE}</span> <span class="va">${ONECHROM}</span> <span class="op">&gt;</span>  <span class="va">${ONECHROM_OR_COMPLETE}</span>

<span class="co">## Count the number of selected species</span>
<span class="fu">wc</span> -l <span class="va">${COMPLETE}</span> <span class="va">${ONECHROM}</span> <span class="va">${ONECHROM_OR_COMPLETE}</span> <span class="va">${ASSEMBLY}</span> </code></pre></div>
<p>The assembly summary also contains information about the status of the genome for the biologist community, with three possible statuses: <em>reference genome</em>, <em>representative genome</em>, or <em>na</em> (non attributed).</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">REFERENCE=</span>ncbi_organisms_assembly_summary_<span class="va">${GROUP}</span>_reference_<span class="va">${TODAY}</span>.tsv
<span class="va">REPRESENTATIVE=</span>ncbi_organisms_assembly_summary_<span class="va">${GROUP}</span>_representative_<span class="va">${TODAY}</span>.tsv
<span class="va">REPRESENTATIVE_COMPLETE=</span>ncbi_organisms_assembly_summary_<span class="va">${GROUP}</span>_representative_complete_<span class="va">${TODAY}</span>.tsv


<span class="co">## Select reference genomes</span>
<span class="fu">awk</span> -F <span class="st">&quot;\t&quot;</span> <span class="st">&#39;$5==&quot;reference genome&quot; &amp;&amp; $11==&quot;latest&quot;{print $0}&#39;</span> \
      <span class="va">${ASSEMBLY}</span> \
      <span class="op">&gt;</span> <span class="va">${REFERENCE}</span>

<span class="co">## Select representative genomes</span>
<span class="fu">awk</span> -F <span class="st">&quot;\t&quot;</span> <span class="st">&#39;$5==&quot;representative genome&quot; &amp;&amp; $11==&quot;latest&quot;{print $0}&#39;</span> \
      <span class="va">${ASSEMBLY}</span> \
      <span class="op">&gt;</span> <span class="va">${REPRESENTATIVE}</span>

<span class="co">## Select representative and complete genomes</span>
<span class="fu">awk</span> -F <span class="st">&quot;\t&quot;</span> <span class="st">&#39;$5==&quot;representative genome&quot; &amp;&amp; $12==&quot;Complete Genome&quot;  &amp;&amp; $11==&quot;latest&quot;{print $0}&#39;</span> \
      <span class="va">${ASSEMBLY}</span> \
      <span class="op">&gt;</span> <span class="va">${REPRESENTATIVE_COMPLETE}</span>

<span class="co">## Print the umbers of reference or representative genomes</span>
<span class="fu">wc</span> -l <span class="va">${REFERENCE}</span> <span class="va">${REPRESENTATIVE_COMPLETE}</span> <span class="va">${REPRESENTATIVE}</span> <span class="va">${COMPLETE}</span> <span class="va">${ONECHROM}</span> <span class="va">${ONECHROM_OR_COMPLETE}</span> <span class="va">${ASSEMBLY}</span> </code></pre></div>
<hr />
</div>
<div id="installing-many-genomes" class="section level2">
<h2><span class="header-section-number">3.5</span> Installing many genomes</h2>
<p>We can install multiple species in a single command by specifying a file containing a list of species names (one species name per row).</p>
<div id="download" class="section level3">
<h3><span class="header-section-number">3.5.1</span> Download</h3>
<p><strong>Beware</strong>:</p>
<ul>
<li><p>This command will download thousands of genomes on your hard drive, which will undoubtedly occupy some disj space.</p></li>
<li><p>The download can take a considerable time depending on the number of selected species and bandwidth of your computer.</p></li>
</ul>
</div>
<div id="downloading-genome-files-from-ncbi" class="section level3">
<h3><span class="header-section-number">3.5.2</span> Downloading genome files from NCBI</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## File with the list of species to install</span>
<span class="va">SPECIES_TO_INSTALL=${GROUP}</span>_species_to_install.txt

<span class="co">## Get the two first words of the 8th column as Genus and species names</span>
<span class="fu">cut</span> -f 8 <span class="va">${COMPLETE}</span> \
  <span class="kw">|</span> <span class="fu">sed</span> <span class="st">&#39;s/ /_/g&#39;</span> \
   <span class="op">&gt;</span> <span class="va">${SPECIES_TO_INSTALL}</span>

<span class="co">## Count the number of species to instal</span>
<span class="fu">wc</span> -l <span class="va">${SPECIES_TO_INSTALL}</span>

<span class="co">## Download all the selected species</span>
<span class="ex">install-organism</span> -v 2 -group <span class="va">${GROUP}</span> \
   -species_file <span class="va">${SPECIES_TO_INSTALL}</span> -task download
   

<span class="co">## Check the list of downloaded organisms </span>
<span class="co">## Note: organism = Genus + species + strain</span>
<span class="ex">install-organism</span> -v 2 -group <span class="va">${GROUP}</span> \
   -species_file <span class="va">${SPECIES_TO_INSTALL}</span> -task list</code></pre></div>
</div>
<div id="parsing" class="section level3">
<h3><span class="header-section-number">3.5.3</span> Parsing</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
<span class="co">## Parse all organisms</span>
<span class="ex">install-organism</span> -v 2 -group <span class="va">${GROUP}</span> \
   -species_file <span class="va">${SPECIES_TO_INSTALL}</span> -task list,parse</code></pre></div>
</div>
<div id="parsing-genomes-from-ncbi-files" class="section level3">
<h3><span class="header-section-number">3.5.4</span> Parsing genomes from NCBI files</h3>
<p>Since the number of genomes to install can be important, we will run <code>install-organism</code> with the option <code>-batch</code>, in order to parallelise the installation steps via a job scheduler. This job scheduler must have been previously defined in RSAT configuration files.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ex">install-organism</span> -v 2 -group <span class="va">${GROUP}</span> \
   -species_file <span class="va">${SPECIES_TO_INSTALL}</span> -task parse -batch

<span class="fu">make</span> -f <span class="va">${RSAT}</span>/makefiles/server.mk  watch_jobs</code></pre></div>
</div>
<div id="rsat-installation" class="section level3">
<h3><span class="header-section-number">3.5.5</span> RSAT installation</h3>
</div>
<div id="default-installation-tasks" class="section level3">
<h3><span class="header-section-number">3.5.6</span> Default installation tasks</h3>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
<span class="co">## Run default installation tasks</span>
<span class="co">## </span><span class="al">NOTE</span><span class="co">: if you need to install hundreds of genomes, this can take a while. ## In such case, it is worth testing the option -batch (see below)</span>
<span class="ex">install-organism</span> -v 2 -group <span class="va">${GROUP}</span> \
   -species_file <span class="va">${SPECIES_TO_INSTALL}</span> -task config,list,default</code></pre></div>
</div>
<div id="managing-installation-tasks-with-a-job-scheduler" class="section level3">
<h3><span class="header-section-number">3.5.7</span> Managing installation tasks with a job scheduler</h3>
<p>If your RSAT server has been configured adequately, you can add the option <code>-batch</code> to the commands above in order to send the installation tasks to a job scheduler.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="va">ORG_TO_INSTALL=${GROUP}</span>_organisms_to_install.txt

<span class="co">## Parse all organisms</span>
<span class="ex">install-organism</span> -v 2 -group <span class="va">${GROUP}</span> \
   -species_file <span class="va">${SPECIES_TO_INSTALL}</span> -task list,parse \
  -batch


<span class="co">## Get the list of organisms from the list of parsed species</span>
<span class="ex">install-organism</span> -v 2 -group <span class="va">${GROUP}</span> \
  -species_file <span class="va">${SPECIES_TO_INSTALL}</span> -task list \
  <span class="op">&gt;</span> <span class="va">${ORG_TO_INSTALL}</span>

<span class="co">## Count the number of species and organisms to install</span>
<span class="va">SPECIES_DO_INSTALL_NB=</span><span class="kw">`</span><span class="fu">grep</span> -v <span class="st">&#39;^;&#39;</span> <span class="va">${SPECIES_TO_INSTALL}</span> <span class="kw">|</span> <span class="fu">wc</span> -l  <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{print $1}&#39;</span><span class="kw">`</span>
<span class="va">ORG_DO_INSTALL_NB=</span><span class="kw">`</span><span class="fu">grep</span> -v <span class="st">&#39;^;&#39;</span> <span class="va">${ORG_TO_INSTALL}</span> <span class="kw">|</span> <span class="fu">wc</span> -l <span class="kw">|</span> <span class="fu">awk</span> <span class="st">&#39;{print $1}&#39;</span><span class="kw">`</span>
<span class="bu">echo</span> <span class="st">&quot;To install: </span><span class="va">${ORG_DO_INSTALL_NB}</span><span class="st"> organisms,  </span><span class="va">${SPECIES_DO_INSTALL_NB}</span><span class="st"> distinct species&quot;</span>

<span class="co">## First retrieve all upstream sequences (cannot be done in batch mode)</span>
<span class="ex">install-organism</span> -v 2  \
   -org_file <span class="va">${ORG_TO_INSTALL}</span> -task config,allup<span class="kw">;</span> 
   
<span class="co">## Send the other installation tasks to the job scheduler</span>
<span class="ex">install-organism</span> -v 2 -org_file <span class="va">${ORG_TO_INSTALL}</span>  -task  start_stop,seq_len_distrib,genome_segments,upstream_freq,genome_freq,protein_freq,protein_len,oligos,dyads,fasta_genome,fasta_genome_rm,chrom_sizes,index_bedtools -batch</code></pre></div>
</div>
</div>
<div id="check-the-installation" class="section level2">
<h2><span class="header-section-number">3.6</span> Check the installation</h2>
<p>TO BE WRITTEN</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
