<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Jacques van Helden" />

<meta name="date" content="2018-02-19" />

<title>Comparative genomics</title>

<script src="comparative_genomics_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="comparative_genomics_files/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="comparative_genomics_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="comparative_genomics_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="comparative_genomics_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="comparative_genomics_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="comparative_genomics_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="comparative_genomics_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="comparative_genomics_files/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Comparative genomics</h1>
<h4 class="author"><em>Jacques van Helden</em></h4>
<h4 class="date"><em>2018-02-19</em></h4>

</div>


<div id="genome-wise-comparison-of-protein-sequences" class="section level1">
<h1><span class="header-section-number">1</span> Genome-wise comparison of protein sequences</h1>
<p>In this section, we explain how to use the program <em>genome-blast</em>, which runs sequence similarity search with <em>BLAST</em> in order to detect significant similarities between all the proteins of a set of genomes.</p>
<p>This operation can take time, and the result tables occupy a considerable amount of space on the hard disk. For this reason, the distribution does thus not include the complete comparison of all genomes against all other ones, but is restricted to some model genomes (<em>Escherichia_coli_GCF_000005845.2_ASM584v2</em> versus all Gammaproteobacteria, <em>Saccharomyces cerevisiae</em> against all Fungi, …).</p>
<p>Depending on your organism of interest, you might wish to perform additional comparisons for your own purpose. In this section, we explain how to compute the similiraty tables between a query organism (e.g. <em>Escherichia_coli_GCF_000005845.2_ASM584v2</em>) and a reference taxon (e.g. all Gammaproteobacteria).</p>
<p>In order to install the tables of similarities between gene products in , you need writing permissions in the directory <code>$RSAT/public_html/data</code>. If this is not the case, ask your system administrator to do it for you.</p>
<div id="running-genome-blast-between-two-genomes" class="section level2">
<h2><span class="header-section-number">1.1</span> Running <em>genome-blast</em> between two genomes</h2>
<p>As a first test, we will use to compare all the gene products (proteins) of a query organism (e.g. <em>Escherichia_coli_GCF_000005845.2_ASM584v2</em>) against all the gene products of a reference organism (e.g. <em>Pseudomonas_aeruginosa_GCF_000006765.1_ASM676v1</em>).</p>
<p>This protocol assumes that the two organisms are already installed on your site, as explained in the installation guide.</p>
<p>We will perform in two steps.</p>
<ol style="list-style-type: decimal">
<li><p>Use the program (which is part of the distribution) to create a BLAST-formatted structure (the “database”) with all proteins of the reference organism ().</p></li>
<li><p>Use the program (part of the distribution) to detect similarities between each protein of the query organism () and the reference organism.</p></li>
</ol>
<div id="formatting-the-blast-database" class="section level4">
<h4><span class="header-section-number">1.1.0.1</span> Formatting the BLAST database</h4>
<p>This DB formatting step is very efficient, it should be completed in a few seconds.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="ex">genome-blast</span> -v 1 -task formatdb \</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">      -q Escherichia_coli_GCF_000005845.2_ASM584v2 \</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">      -db Pseudomonas_aeruginosa_GCF_000006765.1_ASM676v1 </a></code></pre></div>
<p>The result is found in the data directory containing . A new directory has been created, which contains the BLAST-formatted database with all the proteins of the reference organism.</p>
<p>ls -ltr $RSAT/data/genomes/Bacillus_subtilis/blastdb</p>
<p>These are binary files, that you should in principle not open as such.</p>
</div>
<div id="searching-similarities" class="section level4">
<h4><span class="header-section-number">1.1.0.2</span> Searching similarities</h4>
<p>The program compares all the sequences of an input set against all the sequences of a database (the one we just created above). The program generates the appropriate command to find the BLAST database directory, and query it with the proteins of the query organism.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="ex">genome-blast</span> -v 1 -task blastall  \</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">      -q Mycoplasma_pneumoniae \</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">       -db Bacillus_subtilis</a></code></pre></div>
<p>This task takes a bit less that one minute for (because we chose a very small genomes), and can take around 10 minutes for medium-sized bacterial genomes ( 4,000 genes).</p>
<p>Note that the command is written in the verbosity message. If you have specific reasons to customize this command, you can adapt it to apply different parameters.</p>
</div>
<div id="searching-reciprocal-similarities" class="section level4">
<h4><span class="header-section-number">1.1.0.3</span> Searching reciprocal similarities</h4>
<p>One classical orthology criterion (which is not perfect but has practical advantages) is to select the bidirectional best hits as candidate orthologs.</p>
<p>For this, we need to run the reciprocal blast, i.e. using as query organism, and as reference organism.</p>
<p>Note that you can run the two BLAST commands ( and ) in a single shot, by specifying multiple tasks for .</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="ex">genome-blast</span> -v 1 -task formatdb,blastall  \</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">       -q Bacillus_subtilis \</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">       -db Mycoplasma_pneumoniae</a></code></pre></div>
<p>We can now perform a quick test: select the bidirectional best hit () for the gene .</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" data-line-number="1">   <span class="ex">get-orthologs</span> -q NP_109706.1 -uth rank 1 -return all \</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">       -org Mycoplasma_pneumoniae -taxon Bacillus_subtilis</a></code></pre></div>
</div>
<div id="applying-genome-blast-between-a-genome-and-a-taxon" class="section level3">
<h3><span class="header-section-number">1.1.1</span> Applying genome-blast between a genome and a taxon</h3>
<p>Generally, we want to compare a query organism to all the organisms of a given taxon (the ). This can be done with the option .</p>
<p>As an example, we will BLAST all the proteins of against all the proteins of each species of .</p>
<p>We can also use the option to activate the reciprocal search: BLAST all gene products of each bacteria of the reference taxon against those of the query organism .</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="ex">genome-blast</span> -v 1 -task formatdb,blastall -reciprocal \</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">       -q Mycoplasma_pneumoniae \</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">       -dbtaxon Mollicutes</a></code></pre></div>
<p>We can now retrieve the orthologs of a gene (e.g. ) in all Mollicutes.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="ex">get-orthologs</span> -q NP_109706.1 -uth rank 1 -return all \</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">       -org Mycoplasma_pneumoniae -taxon Mollicutes</a></code></pre></div>
</div>
</div>
<div id="getting-putative-homologs-orthologs-and-paralogs" class="section level2">
<h2><span class="header-section-number">1.2</span> Getting putative homologs, orthologs and paralogs</h2>
<p>In this section, I will explain how to use the program . This program takes as input one or several query genes belonging to a given organism (the ), and return the genes whose product (peptidic sequence) show significant similarities with the products of the query genes. The primary usage of is thus to return lists of similar genes, not specialy orthologs. Additional criteria can be imposed to infer orthology. In particular, one of the most common criterion is to select . This can be achieved by imposing the rank 1 with the option .</p>
<p>We will illustrate the concept by retrieving the genes whose product is similar to the protein LexA of , in all the Enterobacteriales. We will then refine the query to extract putative orthologs.</p>
<div id="getting-genes-by-similarities" class="section level3">
<h3><span class="header-section-number">1.2.1</span> Getting genes by similarities</h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="ex">get-orthologs</span> -v 1 -org Escherichia_coli_K_12_substr__MG1655_uid57779 \</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">     -taxon Enterobacteriales \</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">     -q lexA -o lexA_all_hits_Enterobacteriales.tab </a></code></pre></div>
<p>The result file is a list of all the Enterobacteriales genes whose product shows some similarity with the LexA protein from <em>E.coli K12</em>.</p>
<pre><code>   ...
   #ref_id ref_org query
   Sde_1787        Saccharophagus_degradans_2-40   b4043
   CPS_0237        Colwellia_psychrerythraea_34H   b4043
   CPS_2683        Colwellia_psychrerythraea_34H   b4043
   CPS_1635        Colwellia_psychrerythraea_34H   b4043
   IL0262  Idiomarina_loihiensis_L2TR      b4043
   ...
   c5014   Escherichia_coli_CFT073 b4043
   c3190   Escherichia_coli_CFT073 b4043
   b4043   Escherichia_coli_K_12_substr__MG1655_uid57779    b4043
   ...</code></pre>
<p>Each similarity is reported by the ID of the gene, the organism to which is belong, and the ID of the query gene. In this case, the third column contains the same ID on all lines: b4043, which is the ID of the gene lexA in . It seems thus poorly informative, but this column becomes useful when several queries are submitted simultaneously.</p>
</div>
<div id="obtaining-information-on-the-blast-hits" class="section level3">
<h3><span class="header-section-number">1.2.2</span> Obtaining information on the BLAST hits</h3>
<p>The program allows to return additional information on the hits. The list of supported return fields is obtained by calling the command with the option . For example, we can ask to return the percentage of identity, the alignment length, the E-value and the rank of each hit.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="ex">get-orthologs</span> -v 1 -org Escherichia_coli_K_12_substr__MG1655_uid57779 \</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">     -taxon Enterobacteriales \</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">     -q lexA -return ident,ali_len,e_value,rank \</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">     -o lexA_all_hits_Enterobacteriales.tab \</a></code></pre></div>
<p>Which gives the following result:</p>
<pre><code>   ...
   #ref_id ref_org query   ident   ali_len e_value rank
   Sde_1787        Saccharophagus_degradans_2-40   b4043   65.33   199     1e-68   1
   CPS_0237        Colwellia_psychrerythraea_34H   b4043   65.69   204     6e-75   1
   CPS_2683        Colwellia_psychrerythraea_34H   b4043   33.94   109     1e-10   2
   CPS_1635        Colwellia_psychrerythraea_34H   b4043   34.12   85      1e-06   3
   IL0262  Idiomarina_loihiensis_L2TR      b4043   66.83   202     1e-75   1
   ...
   c5014   Escherichia_coli_CFT073 b4043   100.00  202     2e-111  1
   c3190   Escherichia_coli_CFT073 b4043   43.33   90      2e-14   2
   b4043   Escherichia_coli_K_12_substr__MG1655_uid57779    b4043   100.00  202     2e-111  1
   ...</code></pre>
<p>Not surprisingly, the answer includes the self-match of lexA (ID b4043) in , with 100% of identify.</p>
</div>
<div id="selecting-bidirectional-best-hits" class="section level3">
<h3><span class="header-section-number">1.2.3</span> Selecting bidirectional best hits</h3>
<p>We can see that the output contains several matches per genome. For instance, there are 3 matches in . If we assume that these similarities reflect homologies, the result contains thus a combination of paralogs and orthologs.</p>
<p>The simplest criterion to select ortholog is that of . We can select BBH by imposing an upper threshold on the rank, with the option .</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb11-1" data-line-number="1">   <span class="ex">get-orthologs</span> -v 1 -org Escherichia_coli_K_12_substr__MG1655_uid57779 \</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">     -taxon Enterobacteriales \</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">     -q lexA -return ident,ali_len,e_value,rank \</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">     -uth rank 1 \</a>
<a class="sourceLine" id="cb11-5" data-line-number="5">     -o lexA_best_hits_Enterobacteriales_bbh.tab</a></code></pre></div>
<p>The result has now been reduced to admit at most one hit per genome.</p>
<pre><code>   ...
   #ref_id ref_org query   ident   ali_len e_value rank
   Sde_1787        Saccharophagus_degradans_2-40   b4043   65.33   199     1e-68   1
   CPS_0237        Colwellia_psychrerythraea_34H   b4043   65.69   204     6e-75   1
   IL0262  Idiomarina_loihiensis_L2TR      b4043   66.83   202     1e-75   1
   ...
   c5014   Escherichia_coli_CFT073 b4043   100.00  202     2e-111  1
   b4043   Escherichia_coli_K_12_substr__MG1655_uid57779    b4043   100.00  202     2e-111  1
   ...
</code></pre>
</div>
<div id="selecting-hits-with-more-stringent-criteria" class="section level3">
<h3><span class="header-section-number">1.2.4</span> Selecting hits with more stringent criteria</h3>
<p>It is well known that the sole criterion of BBH is not sufficient to infer orthology between two genes. In particular, there is a risk to obtain irrelevant matches, due to partial matches between a protein and some spurious domains. To avoid this, we can add a constraint on the percentage of identity (min 30%), and on the alignment length (min 50 aa). These limits are somewhat arbitrary, we use them to illustrate the principe, and leave to each user the responsibility to choose the criteria that she/he considers as relevant. Finally, we will use a more stringent threshold on E-value than the default one, by imposing an upper threshold of 1e-10.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb13-1" data-line-number="1"></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">   <span class="co">## Note that or this test we suppress the BBH constraint (-uth rank 1)</span></a>
<a class="sourceLine" id="cb13-3" data-line-number="3">   <span class="ex">get-orthologs</span> -v 1 -org Escherichia_coli_K_12_substr__MG1655_uid57779 \</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">     -taxon Enterobacteriales \</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">     -q lexA \</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">     -return ident,ali_len,e_value,rank \</a>
<a class="sourceLine" id="cb13-7" data-line-number="7">     -lth ident 30 -lth ali_len 50 -uth e_value 1e-10 \</a>
<a class="sourceLine" id="cb13-8" data-line-number="8">     -o lexA_filtered_hits_Enterobacteriales_id30_len50_eval-10.tab</a></code></pre></div>
<p>We can now combine the constrains above with the criterion of BBH.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb14-1" data-line-number="1">   <span class="co">## Note that or this test we include the BBH constraint (-uth rank 1)</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2">   <span class="ex">get-orthologs</span> -v 1 -org Escherichia_coli_K_12_substr__MG1655_uid57779 \</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">     -taxon Enterobacteriales \</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">     -q lexA \</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">     -return ident,ali_len,e_value,rank \</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">     -lth ident 30 -lth ali_len 50  -uth e_value 1e-10 \</a>
<a class="sourceLine" id="cb14-7" data-line-number="7">     -uth rank 1 \</a>
<a class="sourceLine" id="cb14-8" data-line-number="8">     -o lexA_BBH_Enterobacteriales_bbh_id30_len50_eval-10.tab</a></code></pre></div>
<p>As expected, the number of selected hits is reduced by adding these constraints. In Sept 2006, we obtained the following number of hits for lexA in Enterobacteriales.</p>
<ul>
<li><p>122 hits without any constraint;</p></li>
<li><p>107 hits with contrains on ident,ali_len and e_value;</p></li>
<li><p>69 hits with the constraint of BBH;</p></li>
<li><p>69 hits with the combined constraint of BBH, at least 30% identity and an alignment over more than 50 aminoacids, and an E-value &lt;= 1.e-10.</p></li>
</ul>
<p>Actually, in the particular case of , the BBH constraint already filtered out the spurious matches, but inother cases they can be useful.</p>
</div>
<div id="restricting-the-number-of-reference-organisms" class="section level3">
<h3><span class="header-section-number">1.2.5</span> Restricting the number of reference organisms</h3>
<p>The decrease of sequencing cost impulsed the multiplication of genome sequencing projects. In 2015, some bacterial species are represented by several hundreds of strains in the EnsemblGenomes database (e.g. ). A conseuence is that the comparative genomics analyses can become quite heavy. A more importance drawback is that some taxonomic branches are over-represented relative to other ones.</p>
<p>To reduce this problem, we added an option to the taxonomic tools.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb15-1" data-line-number="1">   <span class="co">## Note that or this test we include the BBH constraint (-uth rank 1)</span></a>
<a class="sourceLine" id="cb15-2" data-line-number="2">   <span class="ex">get-orthologs</span> -v 1 -org Escherichia_coli_K_12_substr__MG1655_uid57779 \</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">     -taxon Enterobacteriales \</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">     -q lexA -o lexA_orthologs_Enterobacteriales_bbh_id30_len50_eval-10.tab \</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">     -return ident,ali_len,e_value,rank \</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">     -lth ident 30 -lth ali_len 50  -uth e_value 1e-10 \</a>
<a class="sourceLine" id="cb15-7" data-line-number="7">     -uth rank 1 </a></code></pre></div>
</div>
</div>
<div id="retrieving-sequences-for-multiple-organisms" class="section level2">
<h2><span class="header-section-number">1.3</span> Retrieving sequences for multiple organisms</h2>
<p>The program can be used to retrieve sequences for a group of genes belonging to different organisms.This program takes as input a file with two columns. Each row of this file specifies one query gene.</p>
<ol style="list-style-type: decimal">
<li><p>The first column contains the name or identifier of the gene (exactly as for the single-genome program ).</p></li>
<li><p>The second column indicates the organism to which the gne belongs.</p></li>
</ol>
<p>The output of can thus directly be used as input for .</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb16-1" data-line-number="1">   <span class="ex">retrieve-seq-multigenome</span> -noorf \</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">     -i lexA_orthologs_Enterobacteriales_bbh_id30_len50_eval-10.tab \</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">     -o lexA_orthologs_Enterobacteriales_up-noorf.fasta</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">   <span class="kw">\</span></a></code></pre></div>
</div>
<div id="detection-of-phylogenetic-footprints" class="section level2">
<h2><span class="header-section-number">1.4</span> Detection of phylogenetic footprints</h2>
<p><strong>TO BE WRITTEN</strong></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb17-1" data-line-number="1">   <span class="ex">dyad-analysis</span>  -v 1 \</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">     -i lexA_orthologs_Enterobacteriales_up-noorf.fasta \</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">     -sort -2str -noov -lth occ 1 -lth occ_sig 0 \</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">     -return occ,freq,proba,rank \</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">     -l 3 -spacing 0-20 -bg monads \</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">     -o lexA_orthologs_Enterobacteriales_up-noorf_dyads-2str-noov.tab</a></code></pre></div>
</div>
<div id="phylogenetic-profiles" class="section level2">
<h2><span class="header-section-number">1.5</span> Phylogenetic profiles</h2>
<p>The notion of was introduced by Pellegrini et al. (1999). They identified putative orthologs for all the genes of in all the complete genomes available at that time, and built a table with one row per gene, one column per genome. Each cell of this table indicates if an ortholog of the considered gene (row) has been identified in the considered genome (column). Pellegrini et al. (1999) showed that genes having similar phylogenetic profiles are generally involved in common biological processes. The analysis of phylogenetic profiles is thus a powerful way to identify functional grouping in completely sequenced genomes.</p>
<p>The program can be used to obtain the phylogenetic profiles. The principle is to submit the complete list of protein-coding genes of the query organism. We process in two steps :</p>
<ol style="list-style-type: decimal">
<li><p>With , we can identify the putative ortholgos for all the genes of the query organism, using the criterion of . This generate a large table with one row per pair of putative orthologs.</p></li>
<li><p>We then use to convert the ortholog table into profiles (one row per gene, one column per genome).</p></li>
</ol>
<p>We will illustrate this by calculating the phylogenetic profiles of all the genes from across all the Fungi. We use a level of verbosity of 2, in order to get information about the progress of the calculations.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="co">## Identify all the putative orthologs (BBH)     </span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="ex">get-orthologs</span> -v 2 \</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">      -i <span class="va">$RSAT</span>/data/genomes/Saccharomyces_cerevisiae/genome/cds.tab  \</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">      -org Saccharomyces_cerevisiae \</a>
<a class="sourceLine" id="cb18-5" data-line-number="5">      -taxon Fungi \</a>
<a class="sourceLine" id="cb18-6" data-line-number="6">      -uth rank 1 -lth ali_len 50 -lth ident 30 -uth e_value 1e-10 \</a>
<a class="sourceLine" id="cb18-7" data-line-number="7">      -return e_value,bit_sc,ident,ali_len \</a>
<a class="sourceLine" id="cb18-8" data-line-number="8">      -o Saccharomyces_cerevisiae_vs_Fungi_bbh.tab </a>
<a class="sourceLine" id="cb18-9" data-line-number="9"></a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="co">## Convert ortholog table into a profile table     </span></a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="co">## with the IDs of the putative orthologs</span></a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="ex">convert-classes</span> -v 2 \</a>
<a class="sourceLine" id="cb18-13" data-line-number="13">      -i Saccharomyces_cerevisiae_vs_Fungi_bbh.tab  \</a>
<a class="sourceLine" id="cb18-14" data-line-number="14">      -from tab -to profiles \</a>
<a class="sourceLine" id="cb18-15" data-line-number="15">      -ccol 2 -mcol 3 -scol 1 -null <span class="st">&quot;&lt;NA&gt;&quot;</span> \</a>
<a class="sourceLine" id="cb18-16" data-line-number="16">      -o Saccharomyces_cerevisiae_vs_Fungi_phyloprofiles_ids.tab </a></code></pre></div>
<p>The resulting table indicates the identifier of the ortholog genes. The option was used to specify that the string <code>&lt;NA&gt;</code> should be used to indicate the absence of putative orhtolog.</p>
<p>Another option would be to obtain a “quantitative” profile, where each cell indicates the E-value of the match between the two orthologs. This can be done by specifying a different score column with the option of .</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="co">## Convert ortholog table into a profile table</span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">## with the E-value of the putative orthologs</span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="ex">convert-classes</span> -v 2 \</a>
<a class="sourceLine" id="cb19-4" data-line-number="4">      -i Saccharomyces_cerevisiae_vs_Fungi_bbh.tab  \</a>
<a class="sourceLine" id="cb19-5" data-line-number="5">      -from tab -to profiles \</a>
<a class="sourceLine" id="cb19-6" data-line-number="6">      -ccol 2 -mcol 3 -scol 4 -null <span class="st">&quot;&lt;NA&gt;&quot;</span> \</a>
<a class="sourceLine" id="cb19-7" data-line-number="7">      -o Saccharomyces_cerevisiae_vs_Fungi_phyloprofiles_evalue.tab </a></code></pre></div>
</div>
<div id="detecting-pairs-of-genes-with-similar-phylogenetic-profiles" class="section level2">
<h2><span class="header-section-number">1.6</span> Detecting pairs of genes with similar phylogenetic profiles</h2>
<p>In the previous section, we generated tables indicating the phylogenetic profiles of each gene from . This table contains one row per gene, and one column per fungal genome.</p>
<p>We will now use the program to compare each gene profile to each other, to select the pairs of genes with significantly similar profiles. The problem is of course to choose our criterion of similarity between two gene profiles.</p>
<div id="comparing-binary-profiles-with" class="section level3">
<h3><span class="header-section-number">1.6.1</span> Comparing binary profiles with</h3>
<p>For the binary profiles, the most relevant statistics is the .</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="co">## Compare the binary phylogenetic profiles</span></a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="co">## using the hypergeometric significance</span></a>
<a class="sourceLine" id="cb20-3" data-line-number="3"><span class="ex">compare-profiles</span> -v 2 \</a>
<a class="sourceLine" id="cb20-4" data-line-number="4">      -i Saccharomyces_cerevisiae_vs_Fungi_phyloprofiles_evalue.tab \</a>
<a class="sourceLine" id="cb20-5" data-line-number="5">      -lth AB 1 -lth sig 0 \</a>
<a class="sourceLine" id="cb20-6" data-line-number="6">      -return counts,jaccard,hyper,entropy \</a>
<a class="sourceLine" id="cb20-7" data-line-number="7">      -o Saccharomyces_cerevisiae_vs_Fungi_phyloprof_gene_pairs.tab </a></code></pre></div>
<p>In the previous commands, we set the verbosity to 2, in order to keep track the progress of the task. Actually, the processing can take a few minuts, it is probably the good moment for a coffee break.</p>
</div>
<div id="comparing-binary-profiles-with-1" class="section level3">
<h3><span class="header-section-number">1.6.2</span> Comparing binary profiles with</h3>
<p>Another way to compare the phylogenetic profiles is to directly analyze with the table of orthology (previously obtained from ).</p>
<p>This is just another way of considering the same problem: in order to compare genes <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span>, we will consider as a first class (<span class="math inline">\(Q\)</span>) the set of genomes in which gene <span class="math inline">\(A\)</span> is present, and as a second class (<span class="math inline">\(R\)</span>) the set of genomes in which gene <span class="math inline">\(B\)</span> is present. We will then calculate the intersection between these two classes, and assess the significance of this intersection, given the total number of genomes.</p>
<p>Thus, will calculate the hypergeometric statistics, exactly in the same way as .</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="co">## Convert the orthology into &quot;classes&quot;, where each class (second column)   ## corresponds to a gene from Saccharomyces cerevisiae, and indicates      </span></a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="co">## the set of genomes (first column) in which this gene is present.      </span></a>
<a class="sourceLine" id="cb21-3" data-line-number="3"><span class="ex">convert-classes</span> -from tab -to tab -mcol 2 -ccol 3 -scol 5 \</a>
<a class="sourceLine" id="cb21-4" data-line-number="4">      -i Saccharomyces_cerevisiae_vs_Fungi_bbh.tab \</a>
<a class="sourceLine" id="cb21-5" data-line-number="5">      -o Saccharomyces_cerevisiae_vs_Fungi_bbh_classes.tab </a>
<a class="sourceLine" id="cb21-6" data-line-number="6"></a>
<a class="sourceLine" id="cb21-7" data-line-number="7"><span class="co">## Compare the classes to detect significant overlaps</span></a>
<a class="sourceLine" id="cb21-8" data-line-number="8"><span class="ex">compare-classes</span> -v 3 \</a>
<a class="sourceLine" id="cb21-9" data-line-number="9">      -i Saccharomyces_cerevisiae_vs_Fungi_bbh_classes.tab \</a>
<a class="sourceLine" id="cb21-10" data-line-number="10">      -lth QR 1 -lth sig 0 -sort sig -sc 3 \</a>
<a class="sourceLine" id="cb21-11" data-line-number="11">      -return occ,proba,dotprod,jac_sim,rank \</a>
<a class="sourceLine" id="cb21-12" data-line-number="12">      -o phyloprof_gene_pairs.tab </a></code></pre></div>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
